helmfile/charts/networkpolicy-service/templates/cert-manager/certmanager.yaml

```
@@ -0,0 +1,79 @@
+{{ if .Values.certManager.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: cert-manager
+  namespace: cert-manager
+spec:
+  podSelector:
+    matchLabels:
+      app: cert-manager
+  policyTypes:
+    - Ingress
+    - Egress
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchExpressions:
+              - key: kubernetes.io/metadata.name
+                operator: In
+                values:
+                  - monitoring
+          podSelector:
+            matchLabels:
+              app.kubernetes.io/instance: kube-prometheus-stack-prometheus
+      ports:
+        - port: 9402 # promtheus metrics
+  egress:
+    - ports:
+      - port: 53
+        protocol: UDP
+    - to:
+      {{- range $IP := .Values.certManager.letsencrypt.ips }}
+      - ipBlock:
+          cidr: {{ $IP }}
+      {{- end }}
+      ports:
+        - protocol: TCP
+          port: {{ .Values.certManager.letsencrypt.port }}
+    - to:
+        {{- range $IP := .Values.global.scApiserver.ips }}
+        - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      ports:
+        - protocol: TCP
+          port: {{ .Values.global.scApiserver.port }}
+{{- if .Values.global.externalLoadBalancer }}
+    - to:
+        {{- range $IP := .Values.global.scIngress.ips }}
+        - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      ports:
+        - protocol: TCP
+          port: 443
+        - protocol: TCP
+          port: 80
+{{- else if .Values.global.ingressUsingHostNetwork }}
+    - to:
+        {{- range $IP := .Values.global.scNodes.ips }}
+        - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      ports:
+        - protocol: TCP
+          port: 80
+        - protocol: TCP
+          port: 443
+{{ else }}
+    - to: #TODO hostnetwork
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: ingress-nginx
+          podSelector:
+            matchLabels:
+              app.kubernetes.io/component: controller
+              app.kubernetes.io/instance: ingress-nginx
+{{- end }}
```

Nit: some refactoring. You can do similar in other files
```suggestion
    - to:
      {{- if .Values.global.externalLoadBalancer }}
        {{- range $IP := .Values.global.scIngress.ips }}
        - ipBlock:
            cidr: {{ $IP }}
        {{- end }}
      {{- else if .Values.global.ingressUsingHostNetwork }}
        {{- range $IP := .Values.global.scNodes.ips }}
        - ipBlock:
            cidr: {{ $IP }}
        {{- end }}
      {{ else }}
        - namespaceSelector:
            matchLabels:
              kubernetes.io/metadata.name: ingress-nginx
          podSelector:
            matchLabels:
              app.kubernetes.io/component: controller
              app.kubernetes.io/instance: ingress-nginx
      {{- end }}
      ports:
        - protocol: TCP
          port: 443
        - protocol: TCP
          port: 80
```

---

The indentation of the helmfile templating (the if, else, end, ...) is still not fixed here and you can also fix it in several other files.